be/src/common/daemon.cpp

```diff
@@ -111,17 +112,27 @@ void* memory_maintenance_thread(void* dummy) {
  * this thread will calculate some metrics at a fix interval(15 sec)
  * 1. push bytes per second
  * 2. scan bytes per second
+ * 3. max io util of all disks
+ * 4. max network send bytes rate
+ * 5. max network receive bytes rate
  */
 void* calculate_metrics(void* dummy) {
     int64_t last_ts = -1L;
     int64_t lst_push_bytes = -1;
     int64_t lst_query_bytes = -1;
+    std::map<std::string, int64_t> lst_disks_io_time;
+    std::map<std::string, int64_t> lst_net_send_bytes;
+    std::map<std::string, int64_t> lst_net_receive_bytes;
```

Comment:
you can put these map in `while` block to avoid clear

Reply:
The values in these maps need to be used at next loop, so they can not be put in the while block.